impeller/renderer/backend/metal/surface_mtl.h

```diff
@@ -58,6 +59,14 @@ class SurfaceMTL final : public Surface {
   // Returns a Rect defining the area of the surface in device pixels
   IRect coverage() const;
 
+  using DeferredSubmit = std::function<bool()>;
+  using SubmitReciever = std::function<void(DeferredSubmit)>;
+
+  void SetSubmitInfo(
+      const SubmitReciever& submit_reciever) {
+    submit_reciever_ = submit_reciever;
+  }
+
```

Comment:
The usage of callbacks here is due to the fact that impeller::Surface can't really import flow types, so we need to either handle the adaptor in gpu_surface or use something structural. I'll follow up next week with @chinmaygarde to find a better solution for deferring submit.